clang/include/clang/AST/CanonBounds.h

```diff
@@ -118,6 +118,9 @@ namespace clang {
     /// \brief Lexicographic comparison of expressions that can occur in
     /// bounds expressions.
     Result CompareExpr(const Expr *E1, const Expr *E2);
+    /// \brief Semantic comparison of expressions that can occur in
+    /// bounds expressions.
+    bool CompareExprSemantically(const Expr *E1, const Expr *E2);
```

Comment:
You might want to describe the meaning of the return value: that true means two expressions are equivalent semantically.   Clang has this unusual convention where the boolean returns from many functions are inverted in their meaning.